src/Interpreters/MySQL/InterpretersMySQLDDLQuery.cpp

```diff
@@ -117,6 +118,37 @@ static inline NamesAndTypesList getColumnsList(ASTExpressionList * columns_defin
     return columns_name_and_type;
 }
 
+static ColumnsDescription getColumnsDescription(const NamesAndTypesList & columns_name_and_type, const ASTExpressionList * columns_definition)
+{
+    if (columns_name_and_type.size() != columns_definition->children.size())
+            throw Exception("Columns of different size provided.", ErrorCodes::BAD_ARGUMENTS);
+
+    ColumnsDescription columns_description;
+    ColumnDescription column_description;
+
+    for (
+        auto [column_name_and_type, declare_column_ast] = std::tuple{columns_name_and_type.begin(), columns_definition->children.begin()};
+        column_name_and_type != columns_name_and_type.end();
+        column_name_and_type++,
+        declare_column_ast++
+    )
+    {
+        const auto & declare_column = (*declare_column_ast)->as<MySQLParser::ASTDeclareColumn>();
+        String comment;
+        if (declare_column->column_options)
+            if (const auto * options = declare_column->column_options->as<MySQLParser::ASTDeclareOptions>())
+                if (options->changes.count("comment"))
+                    comment = options->changes.at("comment")->as<ASTLiteral>()->value.safeGet<String>();
+
+        column_description.name = column_name_and_type->name;
+        column_description.type = column_name_and_type->type;
+        column_description.comment = std::move(comment);
```

Comment:
`column_description` will have a defined `comment` even if there is no column comment (comment with empty string). Probably it will be better to add it only if it is actually needed.

Reply:
`column_description.comment` is `String`, so it still would be empty string even if we don't explicitly move it there, or doesn't that matter?

Reply:
No, it doesn't.